pandas/tests/test_multilevel.py

```diff
@@ -2781,3 +2781,20 @@ def test_sort_index_nan(self):
         result = s.sort_index(na_position='first')
         expected = s.iloc[[1, 2, 3, 0]]
         tm.assert_series_equal(result, expected)
+
+    def test_sort_ascending_list(self):
+        # GH: 16934
+
+        # Set up a Series with a three level MultiIndex
+        arrays = [['bar', 'bar', 'baz', 'baz', 'foo', 'foo', 'qux', 'qux'],
+                  ['one', 'two', 'one', 'two', 'one', 'two', 'one', 'two'],
+                  [4, 3, 2, 1, 4, 3, 2, 1]]
+        tuples = list(zip(*arrays))
+        index = pd.MultiIndex.from_tuples(tuples,
+                                          names=['first', 'second', 'third'])
+        s = pd.Series(range(8), index=index)
+
+        result = s.sort_index(level=['third', 'first'],
+                              ascending=[False, True])
+
+        assert np.array_equal(result, [0, 4, 1, 5, 2, 6, 3, 7])
```

Comment:
can you also both cases from the original issue (in same tests)

IOW assert first one works as well
```
# Sort descending works
s.sort_index(level=['third', 'first'], ascending=False)

# But sort descending on third level, ascending on first does not
s.sort_index(level=['third', 'first'], ascending=[False, True])
```